### PR TITLE
feat(gestures): port Drag Gesture from use-gesture

### LINKED ISSUE #91: [feat] add use-gesture/drag
https://use-gesture.netlify.app/docs/#vanilla-javascript

### PR DESCRIPTION
**Pull request by nartc**
Closes #91

### CODE REVIEW

File: libs/ngxtension/if-validator/src/if-validator.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import {
 	AbstractControl,
-	AsyncValidatorFn,
 	FormControl,
-	ValidatorFn,
 	Validators,
+	type AsyncValidatorFn,
+	type ValidatorFn,
```
Comment (tomalaforge) [inline libs/ngxtension/if-validator/src/if-validator.ts:6]: nitpick: this refacto should have been done in a separate PR

File: libs/ngxtension/gestures/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export * from './drag';
+export * from './zoneless-gesture';
```
Comment (tomalaforge) [inline libs/ngxtension/gestures/src/index.ts:2]: personal opinion: I always prefer to explicitly export what I want to export

export {xxx} from '....'

**Comment by tomalaforge** [comment]
you want to implement the other behavior in the same PR?

File: libs/ngxtension/if-validator/src/if-validator.ts (hunk 1 above)
Comment (nartc) [reply libs/ngxtension/if-validator/src/if-validator.ts:6]: Yeah, makes sense. I'll turn on `verbatimModuleSyntax` in a different PR

File: libs/ngxtension/gestures/src/index.ts (hunk 2 above)
Comment (nartc) [reply libs/ngxtension/gestures/src/index.ts:2]: I do it on a case-by-case basis. There's no value to do `export { xxx }` in these cases

**Comment by nartc** [comment]
@tomalaforge I guess it doesn't have to be the same PR. I'll clean up the verbatim tsconfig then will update the PR with tests.

**Comment by eneajaho** [comment]
LGTM! 

Can't wait to have all the helpers in the lib!
